chat/chat.go

```diff
@@ -0,0 +1,91 @@
+// This code as it currently stands is based off
+// https://github.com/garyburd/gary.burd.info/tree/master/go-websocket-chat
+//
+// Naturally, this will change as CPSSDChat becomes more advanced.
+
+package chat
+
+import (
+	"github.com/gorilla/websocket"
+	"log"
+)
+
+type Room struct {
+	// All active connections
+	connections map[*Connection]bool
+
+	// A buffered channel of incoming messages. []byte and not string
+	// becaus the websockets use []bytes.
+	messages chan []byte
+
+	// Requests to join the room
+	Register chan *Connection
+
+	// Requests to leave the room
+	Unregister chan *Connection
+}
+
+func NewRoom() *Room {
+	return &Room{
+		connections: make(map[*Connection]bool),
+		messages:    make(chan []byte, 20),
```

Comment:
For line 31 it means create a channel (basically a queue) buffered to hold 20 byte arrays before blocking.